R/eval.R

```diff
@@ -89,20 +89,24 @@ evaluate <- function(input,
     if (log_echo || debug) {
       cat_line(parsed$src[[i]], file = stderr())
     }
-    evaluate_top_level_expression(
-      exprs = parsed$expr[[i]],
-      src = parsed$src[[i]],
-      watcher = watcher,
-      envir = envir,
-      use_try = on_error != "error",
-      keep_warning = keep_warning,
-      keep_message = keep_message,
-      log_warning = log_warning,
-      output_handler = output_handler
+    continue <- withRestarts(
+      evaluate_top_level_expression(
+        exprs = parsed$expr[[i]],
+        src = parsed$src[[i]],
+        watcher = watcher,
+        envir = envir,
+        on_error = on_error,
+        keep_warning = keep_warning,
+        keep_message = keep_message,
+        log_warning = log_warning,
+        output_handler = output_handler
+      ),
```

Comment:
I'd wrap this in a `{` block with `TRUE` as return value, so it's immediately clear what's going on when `evaluate_top_level_expression()` returns normally.